media-libs/libsdl2/patches/libsdl2-2.0.20.patchset

```diff
@@ -1267,143 +1267,532 @@ index ca9dcad..1ac3201 100644
 2.30.2
 
 
-From 369898e9f76829511e6cdd922866696d414aa963 Mon Sep 17 00:00:00 2001
+From ab6dfb1b1f57e780cab134a67db3600c4e350a5e Mon Sep 17 00:00:00 2001
 From: Ivan Holmes <ivan@ivanholmes.co.uk>
-Date: Sat, 7 May 2022 19:17:29 +0100
-Subject: Don't create a BApplication if one already exists
+Date: Sat, 7 May 2022 22:10:49 +0100
+Subject: Move stuff to devicedata
 
 
-diff --git a/src/main/haiku/SDL_BeApp.cc b/src/main/haiku/SDL_BeApp.cc
-index a5e1428..3b0f784 100644
---- a/src/main/haiku/SDL_BeApp.cc
-+++ b/src/main/haiku/SDL_BeApp.cc
-@@ -84,10 +84,12 @@ SDL_InitBeApp(void)
+diff --git a/src/audio/haiku/SDL_haikuaudio.cc b/src/audio/haiku/SDL_haikuaudio.cc
+index 82da478..47c8837 100644
+--- a/src/audio/haiku/SDL_haikuaudio.cc
++++ b/src/audio/haiku/SDL_haikuaudio.cc
+@@ -220,7 +220,7 @@ static int
+ HAIKUAUDIO_Init(SDL_AudioDriverImpl * impl)
  {
-     /* Create the BApplication that handles appserver interaction */
-     if (SDL_BeAppActive <= 0) {
-+    	if (!be_app) {
-         SDL_AppThread = SDL_CreateThreadInternal(StartBeApp, "SDLApplication", 0, NULL);
-         if (SDL_AppThread == NULL) {
-             return SDL_SetError("Couldn't create BApplication thread");
-         }
-+    	}
+     /* Initialize the Be Application, if it's not already started */
+-    if (SDL_InitBeApp() < 0) {
++    if (SDL_InitBeApp(NULL) < 0) {
```

Comment:
I am pretty sure this will cause a problem (literally passing NULL as the value of a pointer...). I didn't realise that `SDL_InitBeApp` was called in the audio driver code when I made it take the `SDL_VideoDevice` as an argument. Need to find a better way of making the `devicedata` visible to the BApp.